src/types/memory-controller.ts

```diff
@@ -18,8 +34,21 @@ export class MemoryController {
     return new GameData(cards);
   }
 
-  private fetchData(): GameDataDTO {
-    return testData;
+  public async fetchData(): Promise<GameData> {
+    try {
+      const result = await axios.get<GameDataDTO>(
+        `${baseURL}/configurations/${configurationId}`
+      );
+      const gameData = this.convertDTOToData(result.data);
+      this.gameData = gameData;
+      this.shuffleCards();
+      return gameData;
+    } catch (error) {
+      const gameData = this.convertDTOToData(testData);
+      this.gameData = gameData;
+      this.shuffleCards();
+      return gameData;
+    }
```

Comment:
If configuration could not be fetched, it is not the best way to let the user just play memory with dummy data (probably not even topic related). I think the best approach would be to show an error message and do not let the user to play with dummy data.